### PR TITLE
Fix NPD test to run it in private and large clusters

### PR DESCRIPTION
**Pull request by wojtek-t**
```release-note
NONE
```

/kind cleanup
/sig node
/sig scalability
/priority important-longterm

### CODE REVIEW

**Comment by wojtek-t** [comment]
/assign @SergeyKanzhelev 

@SergeyKanzhelev - can you please help with reviewing it?

**Review by vteratipally — APPROVED** [review]

**Comment by wojtek-t** [comment]
/assign @dims 

@dims - can you please help with reviewing?

File: test/e2e/node/node_problem_detector.go [hunk 1]
```diff
@@ -41,10 +41,11 @@ import (
...
 // This test checks if node-problem-detector (NPD) runs fine without error on
 // the nodes in the cluster. NPD's functionality is tested in e2e_node tests.
-var _ = SIGDescribe("NodeProblemDetector [DisabledForLargeClusters]", func() {
+var _ = SIGDescribe("NodeProblemDetector", func() {
 	const (
-		pollInterval = 1 * time.Second
-		pollTimeout  = 1 * time.Minute
+		pollInterval      = 1 * time.Second
+		pollTimeout       = 1 * time.Minute
+		maxNodesToProcess = 10
```
Comment (dims) [inline test/e2e/node/node_problem_detector.go:48]: Do we want to fix the comment in lines 42/43 to say that we will check up-to-max-of-10-nodes

File: test/e2e/node/node_problem_detector.go [hunk 2]
```diff
@@ -60,18 +61,31 @@ var _ = SIGDescribe("NodeProblemDetector [DisabledForLargeClusters]", func() {
...
 					break
 				}
 			}
+			// Not every node has to have an external IP address.
+			if len(host) > 0 {
+				nodes = append(nodes, node)
+				hosts = append(hosts, host)
+			}
+		}
+		framework.ExpectEqual(len(hosts), len(nodes))
```
Comment (dims) [inline test/e2e/node/node_problem_detector.go:83]: do we really need this check?

File: test/e2e/node/node_problem_detector.go [hunk 3]
```diff
@@ -60,18 +61,31 @@ var _ = SIGDescribe("NodeProblemDetector [DisabledForLargeClusters]", func() {
...
+
+		if len(nodes) > maxNodesToProcess {
+			nodes = nodes[:maxNodesToProcess]
```
Comment (dims) [inline test/e2e/node/node_problem_detector.go:86]: looks good, we just need to process the first 10.

**Comment by dims** [comment]
@wojtek-t tiny nit/question, LGTM! 

Looks like for a cluster with a *LOT* of nodes, we want to check only the first 10 (that have an external IP ... and we pick only one external IP per node as well.)

**Comment by dims** [comment]
/lgtm
/hold

feel free to remove hold if you just want to merge this as-is. happy to re-lgtm if you want to fix anything @wojtek-t

File: test/e2e/node/node_problem_detector.go (hunk 1 above)
Comment (wojtek-t) [reply test/e2e/node/node_problem_detector.go:48]: done

File: test/e2e/node/node_problem_detector.go (hunk 2 above)
Comment (wojtek-t) [reply test/e2e/node/node_problem_detector.go]: removed

**Review by wojtek-t — COMMENTED** [review]
@dims - thanks a lot, comments applied, PTAL when you will have a chance

File: test/e2e/node/node_problem_detector.go [hunk 4]
```diff
@@ -60,18 +61,30 @@ var _ = SIGDescribe("NodeProblemDetector [DisabledForLargeClusters]", func() {
...
 				}
 			}
+			// Not every node has to have an external IP address.
+			if len(host) > 0 {
+				nodes = append(nodes, node)
+				hosts = append(hosts, host)
+			}
+		}
+
+		if len(nodes) > maxNodesToProcess {
```
Comment (SergeyKanzhelev) [inline test/e2e/node/node_problem_detector.go:84]: should this check that we got at least one node in the list?

Comment (SergeyKanzhelev) [reply test/e2e/node/node_problem_detector.go:84]: /hold 

for this comment.

Comment (wojtek-t) [reply test/e2e/node/node_problem_detector.go:84]: > should this check that we got at least one node in the list?

What should we do if there isn't? Technically, it's a completely valid setup for a node to not have public IP. So in that case, this test is incompatible with that. But we won't fail the test anyway - we may potentially log something, but not sure how much it helps us...

Comment (SergeyKanzhelev) [reply test/e2e/node/node_problem_detector.go:84]: in case there is no valid IP test doesn't test anything. Maybe `ginkgo.Skip` as failed prerequisites?

Comment (wojtek-t) [reply test/e2e/node/node_problem_detector.go:88]: done

**Comment by wojtek-t** [comment]
@SergeyKanzhelev - PTAL

**Comment by SergeyKanzhelev** [comment]
/hold

to stop auto-retest

**Comment by wojtek-t** [comment]
Fixed the typo - reapplying lgtm. Thanks!

**Pull request merged**